File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestListOptStringValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.test.TypeTester;
import com.dslplatform.ocd.values.ListOptStringInValue.ListOptStringValue;
import com.dslplatform.patterns.ServiceLocator;
import java.util.*;
import org.junit.*;
import static org.junit.Assert.*;

public class TestListOptStringValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
        locator = Bootstrap.init(TestListOptStringValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertTrue(TypeTester.testField(ListOptStringValue.class, "listOptString")
                .resultEquals(List.class, String.class));
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testGetter(ListOptStringValue.class, "getListOptString")
                .resultEquals(List.class, String.class));
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testSetter(ListOptStringValue.class, "setListOptString", List.class, String.class)
                .resultEquals(ListOptStringValue.class));
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestListOptStringValue.scala
package com.dslplatform.ocd.values

import ListOptStringInValue.ListOptStringValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestListOptStringValue extends SpecExtensions {
  def `Test property field type: IndexedSeq[Option[String]]`() =
    checkType(
      typeOf[IndexedSeq[Option[String]]]
    , typeOf[ListOptStringValue].member("listOptString": TermName).asMethod.returnType
    )

  def `Test property default value: IndexedSeq.empty[Option[String]]`() {
    assert(ListOptStringValue().listOptString === IndexedSeq.empty[Option[String]])
  }
}
